src/kernels/kernel.ts

```diff
@@ -536,16 +534,18 @@ abstract class BaseKernel<TKernelExecution extends BaseKernelExecution> implemen
         // Gather all of the startup code into a giant string array so we
         // can execute it all at once.
         const result: string[] = [];
-        if (isPythonKernelConnection(this.kernelConnectionMetadata)) {
-            const dirs = await Promise.all(
-                this.startupCodeProviders
-                    .sort((a, b) => b.priority - a.priority)
-                    .map((provider) => provider.getCode(this))
-            );
-            for (let dir of dirs) {
-                result.push(...dir);
-            }
+        const startupCode = await Promise.all(
+            this.startupCodeProviders.sort((a, b) => b.priority - a.priority).map((provider) => provider.getCode(this))
+        );
+        for (let code of startupCode) {
+            result.push(...code);
+        }
 
+        // If this is a live kernel, we shouldn't be changing anything by running startup code.
```

Comment:
we've already gathered startup code from the injected providers, so this comment isn't accurate.

Reply:
Nope it is still correct, if you check the code in the if condition, we're actually injecting some code, and that should be done only for non-live kernels. ie. the comment is required.